crates/sui-types/src/crypto.rs

```diff
@@ -508,19 +508,38 @@ impl PartialEq for AuthoritySignInfo {
 }
 
 impl AuthoritySignInfo {
-    pub fn add_to_verification_obligation(
+    pub fn add_to_verification_obligation<T>(
         &self,
+        data: &T,
         committee: &Committee,
         obligation: &mut VerificationObligation,
-        message_index: usize,
-    ) -> SuiResult<()> {
+    ) -> SuiResult<()>
+    where
+        T: Signable<Vec<u8>>,
+    {
+        let weight = committee.weight(&self.authority);
+        fp_ensure!(weight > 0, SuiError::UnknownSigner);
+
         obligation
             .public_keys
             .push(committee.public_key(&self.authority)?);
         obligation.signatures.push(self.signature.0);
+        let mut message = Vec::new();
```

Comment:
I think this works fine for now, as we are using Ed25519 signatures, but in an upcoming PR where I make the VerificationObligation agnostic to the type of signature (including aggregated signatures), this wouldn't work. 

Ideally, when adding to the VerificationObligation, we want all signatures that are signed on top of one message, to refer to one message idx in the VerificationObligation - not only would this (very slightly) reduce the amount of messages that we write to memory, but it will allow us to aggregate all signatures that refer to one message, if we are using an aggregatable signature scheme.

Reply:
By no means am I saying that we are changing the signature scheme, but I feel like this would be more intuitive, and would be more flexible for experimenting with future crypto protocols, especially if we add multisigs etc.

Reply:
Sounds good!